apis/network/v1alpha3/zone_types.go

```diff
@@ -0,0 +1,130 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	runtimev1alpha1 "github.com/crossplane/crossplane-runtime/apis/core/v1alpha1"
+)
+
+// +kubebuilder:object:root=true
+
+// Zone is a managed resource that represents an AWS Route53 Hosted Zone.
+// +kubebuilder:printcolumn:name="ID",type="string",JSONPath=".status.atProvider.Id"
+// +kubebuilder:printcolumn:name="RRs",type="integer",JSONPath=".status.atProvider.ResourceRecordSetCount"
+// +kubebuilder:printcolumn:name="READY",type="string",JSONPath=".status.conditions[?(@.type=='Ready')].status"
+// +kubebuilder:printcolumn:name="SYNCED",type="string",JSONPath=".status.conditions[?(@.type=='Synced')].status"
+// +kubebuilder:printcolumn:name="AGE",type="date",JSONPath=".metadata.creationTimestamp"
+// +kubebuilder:subresource:status
+// +kubebuilder:resource:scope=Cluster
+type Zone struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ZoneSpec   `json:"spec"`
+	Status ZoneStatus `json:"status,omitempty"`
+}
+
+// ZoneSpec defines the desired state of an AWS Route53 Hosted Zone.
+type ZoneSpec struct {
+	runtimev1alpha1.ResourceSpec `json:",inline"`
+	ForProvider                  ZoneParameters `json:"forProvider"`
+}
+
+// ZoneStatus represents the observed state of a Zone.
+type ZoneStatus struct {
+	runtimev1alpha1.ResourceStatus `json:",inline"`
+	AtProvider                     ZoneObservation `json:"atProvider"`
+}
+
+// ZoneParameters define the desired state of an AWS Route53 Hosted Zone.
+type ZoneParameters struct {
```

Comment:
Looking at `CreateHostedZone`, `HostedZoneConfig` seems to be a sub-struct of the main object but here the properties of `HostedZoneConfig` like `Comment` and `PrivateZone` are at the top level. As per [managed resource design](https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#high-fidelity), we're trying to stick to whatever cloud provider exposes as API, so, it'd be good to follow that.

Reply:
I was told to keep the structs fields around the concrete types as much as possible and hence I choose to do this. I have added `HostedZoneConfig` as suggested.